### PR TITLE
Octopus strip '/' instance URL suffix if provided

### PR DESCRIPTION
**Pull request by willmarks**
## Description

Strip trailing `/` suffix off Octopus instance url if provided.

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/octopus-source/src/octopusClient.ts [hunk 1]
```diff
@@ -45,8 +45,16 @@ export class OctopusClient {
     const retries = config.maxRetries ?? DEFAULT_MAX_RETRIES;
     const logger = this.logger; // for axios-retry
 
+    const instanceUrlRegex = new RegExp(/^((https|http):\/\/[^/]*)\/?$/);
```
Comment (tovbinm) [inline sources/octopus-source/src/octopusClient.ts:48]: perhaps just use this one instead? https://github.com/axios/axios/blob/2e70cecda42993e1153248f0f96715c3c55f7f39/dist/browser/axios.cjs#L2015

File: sources/octopus-source/src/octopusClient.ts [hunk 2]
```diff
@@ -45,8 +45,10 @@
     const retries = config.maxRetries ?? DEFAULT_MAX_RETRIES;
     const logger = this.logger; // for axios-retry
 
+    const cleanInstanceUrl = config.instanceUrl.replace(/\/+$/, '');
```
[the remark this answers is not in the document]
Comment (willmarks) [reply sources/octopus-source/src/octopusClient.ts:48]: @tovbinm Not sure if we care about this, but this is being executed on user input...

[the remark this answers is not in the document]
Comment (tovbinm) [reply sources/octopus-source/src/octopusClient.ts:48]: Then just don't use regex. Strip a single / at then if present

[the remark this answers is not in the document]
Comment (willmarks) [reply sources/octopus-source/src/octopusClient.ts:48]: That will strip the `https://` slashes.

[the remark this answers is not in the document]
Comment (ypc-faros) [reply sources/octopus-source/src/octopusClient.ts:48]: "Then just don't use regex. Strip a single / at then if present"

I think @tovbinm might have meant "Then just don't use regex. Strip a single / at **the end** if present"

**Review by ypc-faros — APPROVED** [review]

**Pull request merged**